apps/challenges/utils.py

```diff
@@ -64,14 +64,58 @@ def convert_to_aws_federated_user_format(string):
     return result
 
 
-def get_or_create_ecr_repository(name, region_name="us-east-1"):
+def get_aws_credentials_for_challenge(challenge_pk):
+    """
+    Return the AWS credentials for a challenge using challenge pk
+    Arguments:
+        challenge_pk {int} -- challenge pk for which credentails are to be fetched
+    Returns:
+        aws_key {dict} -- Dict containing aws keys for a challenge
+    """
+    challenge = get_challenge_model(challenge_pk)
+    if challenge.use_host_credentials:
+        aws_keys = {
+            "AWS_ACCOUNT_ID": challenge.aws_account_id,
+            "AWS_ACCESS_KEY_ID": challenge.aws_access_key_id,
+            "AWS_SECRET_ACCESS_KEY": challenge.aws_secret_access_key,
+            "AWS_REGION": challenge.aws_region,
+        }
+    else:
+        aws_keys = {
+            "AWS_ACCOUNT_ID": os.environ.get("AWS_ACCOUNT_ID"),
+            "AWS_ACCESS_KEY_ID": os.environ.get("AWS_ACCESS_KEY_ID"),
+            "AWS_SECRET_ACCESS_KEY": os.environ.get("AWS_SECRET_ACCESS_KEY"),
+            "AWS_REGION": os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+        }
+    return aws_keys
+
+
+def get_boto3_client(resource, aws_keys):
```

Comment:
Can you please move this to base app since it can be used in other apps and is not tied to `challenges` app?